server.js

```diff
@@ -1,20 +1,114 @@
 const express = require("express");
 const cors = require("cors");
+const moment = require("moment");
+const validator = require("email-validator");
+const fs = require("fs");
 
 const app = express();
-
 app.use(express.json());
 app.use(cors());
 
 //Use this array as your (in-memory) data store.
 const bookings = require("./bookings.json");
 
+
+let Model = [
+  "title",
+  "firstName",
+  "surname",
+  "email",
+  "roomId",
+  "checkInDate",
+  "checkOutDate",
+];
+
+//Helper functions.................................................
+
+function getMaxId(arr) {
+  let idArray = arr.map((el) => (el = el.id));
+  return Math.max(...idArray);
+}
+
+function isValid(obj) {
+  const receivedData = Object.keys(obj);
+  if (Model.every((x) => receivedData.includes(x))) {
+    return validator.validate(obj.email);
+  }
+  return false;
+}
+
+// Controller Functions.....................................
+const queryBookings = (req, res) => {
+  let result = bookings;
+
+  if (req.query.date) {
+    const inputDate = moment(req.query.date);
+    result = bookings.filter((item) =>
+      moment(item.checkInDate).isSame(moment(inputDate))
+    );
+  }
+
+  if (req.query.term) {
+    let term = req.query.term;
+    result = result.filter((item) => Object.values(item).includes(term));
+  }
+  res.json(result);
+};
+
+const getAllBookings = (req, res) => {
+  res.json(bookings);
+};
+
+const getBooking = (req, res) => {
+  const id = req.params.id;
+  const result = bookings.find((item) => item.id === +id);
+  if (result) {
+    res.json(result);
+  } else {
+    res.status(404).send("Not Found");
+  }
+};
+
+const createBooking = (req, res) => {
+  const result = {};
+  const id = getMaxId(bookings) + 1;
+  if (isValid(req.body)) {
+    result.id = id;
+    for (let key of Model) {
+      result[key] = req.body[key];
+    }
+    bookings.push(result);
+    fs.writeFile("bookings.json", JSON.stringify(bookings), (err) =>
```

Comment:
I am trying to send some data to this endpoint, but it doesn't seem to be working for me. 
What result do you get?

>   {
    "title": "Mr",
    "firstName": "Louis",
    "surname": "Boyd",
    "email": "jamesbrown@example.com",
    "roomId": 122,
    "checkInDate": "2018-02-15",
    "checkOutDate": "2018-02-28"
  }

Reply:
I checked it postman now. I got a Json of all the data with the newly created data as the last entry. 
<img width="626" alt="image" src="https://user-images.githubusercontent.com/103941234/213147411-a8a8565e-f99e-4326-86aa-f5ea113044b4.png">